resources/fetchers/kube_fetcher.go

```diff
@@ -145,7 +145,7 @@ func (f *KubeFetcher) initWatchers() error {
 	return nil
 }
 
-func (f *KubeFetcher) Fetch(ctx context.Context) ([]fetching.Resource, error) {
+func (f *KubeFetcher) Fetch(ctx context.Context, resCh chan<- fetching.ResourceInfo, cMetadata fetching.CycleMetadata) error {
```

Comment:
I think a cleaner approach could be:
1. On fetcher creation, give it the result channel
2. My next comment on registry